lib/auth/fp.go

```diff
@@ -0,0 +1,48 @@
+package auth
```

Comment:
I'm not sure the auth package is the right place for these but I hesitate to suggest dumping them into `utils`, perhaps a new package?

at a minimum you'll need a license header here